src/gen/local_lm.py

```diff
@@ -0,0 +1,47 @@
+import torch
+from transformers import (
+    AutoModelForCausalLM,
+    BitsAndBytesConfig
+)
+from alignment.model_utils import get_tokenizer
+
+def get_model(model_id, load_in_8bit, load_in_4bit, model_args, data_args):
+    """
+    get_model instantiates and return fine-tuned language model and tokenzier.
+
+    arguments:
+    model_args -- ModelArguments obtained from H4ArgumentParser
+    data_args -- DataArguments obtained from H4ArgumentParser
+    """
+    quantization_config = BitsAndBytesConfig(load_in_8bit=load_in_8bit, load_in_4bit=load_in_4bit)
+
+    tokenizer = get_tokenizer(model_args, data_args)
+    model = AutoModelForCausalLM.from_pretrained(
+        model_id, torch_dtype=torch.bfloat16,
+        quantization_config=quantization_config, device_map="auto"
+    )
+
+    return tokenizer, model
+
+def gen_model_output(model, tokenizer, ds, temperature=0.4, max_new_tokens=1024, delimiter="assistant\n"):
```

Comment:
Here, if we directly pass / allow a list of prompts, it becomes a batched a implementation.

Reply:
I haven't tried batched inference yet, but is it that simple?
gonna give it a try anyways :)

Reply:
Yeah.